app/org/sagebionetworks/bridge/dynamodb/DynamoSurveyResponse.java

```diff
@@ -1,100 +1,101 @@
-        survey.setHealthCode(JsonUtils.asText(node, HEALTH_CODE_PROPERTY));
-        survey.setVersion(JsonUtils.asLong(node, VERSION_PROPERTY));
-        survey.setStartedOn(JsonUtils.asMillisSinceEpoch(node, STARTED_ON_PROPERTY));
-        survey.setCompletedOn(JsonUtils.asMillisSinceEpoch(node, COMPLETED_ON_PROPERTY));
-        survey.setData((ObjectNode)node);
-        return survey;
-    }
-    
-    private String surveyGuid; // stored in dynamo
-    private long surveyCreatedOn; // stored in dynamo
-    private Survey survey; // constructed and returned to the consumer
-    private String identifier;
     private String healthCode;
-    private long startedOn;
-    private long completedOn;
+    private String identifier;
+    
+    // These three are stored in Dynamo.
+    private String surveyKey;
+    
+    private Long startedOn;
+    private Long completedOn;
     private Long version;
     private List<SurveyAnswer> answers = Lists.newArrayList();
     
-    public DynamoSurveyResponse() {
+    @Override
+    @JsonIgnore
+    @DynamoDBHashKey
+    public String getHealthCode() {
+        return healthCode;
     }
-    
-    public DynamoSurveyResponse(String healthCode, String identifier) {
+    @Override
+    public void setHealthCode(String healthCode) {
         this.healthCode = healthCode;
-        this.identifier = identifier;
     }
     
-    @DynamoDBAttribute
-    @JsonIgnore
-    public String getSurveyGuid() {
-        return surveyGuid;
+    // Main range key: identifier
+    @Override
+    @DynamoDBRangeKey
+    public String getIdentifier() {
+        return identifier;
     }
-    public void setSurveyGuid(String surveyGuid) {
-        this.surveyGuid = surveyGuid;
+    @Override
+    public void setIdentifier(String identifier) {
+        this.identifier = identifier;
     }
-    @DynamoDBAttribute
+    
     @JsonIgnore
-    public long getSurveyCreatedOn() {
-        return surveyCreatedOn;
+    @DynamoDBAttribute
+    @DynamoDBIndexHashKey(globalSecondaryIndexName = "surveyKey-index")
+    public String getSurveyKey() {
+        return surveyKey;
     }
-    public void setSurveyCreatedOn(long surveyCreatedOn) {
-        this.surveyCreatedOn = surveyCreatedOn;
+    public void setSurveyKey(String surveyKey) {
+        this.surveyKey = surveyKey;
     }
-    @Override
-    @JsonIgnore
-    @DynamoDBAttribute
-    public String getHealthCode() {
-        return healthCode;
+    public void setSurveyKey(GuidCreatedOnVersionHolder keys) {
+        this.surveyKey = String.format("%s:%s", keys.getGuid(), DateUtils.convertToISODateTime(keys.getCreatedOn()));
     }
+    @DynamoDBIgnore
+    @JsonIgnore
     @Override
-    public void setHealthCode(String healthCode) {
-        this.healthCode = healthCode;
+    public String getSurveyGuid() {
+        try {
+            return (surveyKey != null) ? surveyKey.split(":")[0] : null;    
+        } catch(Exception e) {
+            return null;
+        }
     }
```

Comment:
Is it worth having a unit test to make sure getSurveyGuid() and getSurveyCreatedOn() correctly parse their fields out of surveyKey?